var/spack/repos/builtin/packages/tcl/package.py

```diff
@@ -11,7 +11,23 @@
 from spack.util.environment import is_system_path
 
 
-class Tcl(AutotoolsPackage, SourceforgePackage):
+class TclMixin:
+    def _find_script_dir(self):
+        # Put more-specific prefixes first
+        check_prefixes = [
+            join_path(
+                self.spec.prefix, "share", "tcl{0}".format(self.spec.package.version.up_to(2))
```

Comment:
It takes some consideration to understand why this works (both Package and Builder have a `.spec`) - I'm generally of the opinion that this would be better-factored as a static function or standalone class called by each scenario, rather than depending on the implicit shared interface between the classes.